Incorrect TypeScript Frontend parser path detection

# Stack trace

*Stack trace in case of an Exception*
```gradle
> Task :run
--- CPG TypeScript Analyzer ---
SLF4J(W): Class path contains multiple SLF4J providers.
SLF4J(W): Found provider [org.slf4j.simple.SimpleServiceProvider@68bbe345]
SLF4J(W): Found provider [org.apache.logging.slf4j.SLF4JServiceProvider@30b8a058]
SLF4J(W): See https://www.slf4j.org/codes.html#multiple_bindings for an explanation.
SLF4J(I): Actual provider is of type [org.slf4j.simple.SimpleServiceProvider@68bbe345]
[main] INFO de.fraunhofer.aisec.cpg.TranslationConfiguration - Passes before enforcing order: []
[main] INFO de.fraunhofer.aisec.cpg.TranslationConfiguration - The following mermaid graph represents the pass 
[main] INFO de.fraunhofer.aisec.cpg.passes.configuration.PassOrderingHelper - Passes after enforcing order: []
[ForkJoinPool.commonPool-worker-1] INFO de.fraunhofer.aisec.cpg.helpers.MeasurementHolder - TranslationManager: Translation into full graph
[ForkJoinPool.commonPool-worker-1] INFO de.fraunhofer.aisec.cpg.helpers.MeasurementHolder - TranslationManager: Executing Language Frontend
[ForkJoinPool.commonPool-worker-1] INFO de.fraunhofer.aisec.cpg.TranslationManager - Parsing secret-leak.ts
[ForkJoinPool.commonPool-worker-1] INFO de.fraunhofer.aisec.cpg.helpers.MeasurementHolder - TranslationManager: Translation into full graph done in 3942 ms
Exception in thread "main" java.util.concurrent.ExecutionException: java.io.IOException: Cannot run program "/tmp/parser12814639995727740945": error=13, Permission denied
	at java.base/java.util.concurrent.CompletableFuture.reportGet(CompletableFuture.java:396)
	at java.base/java.util.concurrent.CompletableFuture.get(CompletableFuture.java:2073)
	at org.example.MainKt.main(Main.kt:36)
	at org.example.MainKt.main(Main.kt)
Caused by: java.io.IOException: Cannot run program "/tmp/parser12814639995727740945": error=13, Permission denied
	at java.base/java.lang.ProcessBuilder.start(ProcessBuilder.java:1170)
Caused by: java.io.IOException: Cannot run program "/tmp/parser12814639995727740945": error=13, Permission denied

	at java.base/java.lang.ProcessBuilder.start(ProcessBuilder.java:1089)
	at java.base/java.lang.Runtime.exec(Runtime.java:681)
	at java.base/java.lang.Runtime.exec(Runtime.java:530)
	at de.fraunhofer.aisec.cpg.frontends.typescript.TypeScriptLanguageFrontend.parse(TypeScriptLanguageFrontend.kt:114)
	at de.fraunhofer.aisec.cpg.TranslationManager.parse(TranslationManager.kt:482)
	at de.fraunhofer.aisec.cpg.TranslationManager.parseSequentially(TranslationManager.kt:434)
	at de.fraunhofer.aisec.cpg.TranslationManager.runFrontends(TranslationManager.kt:256)
	at de.fraunhofer.aisec.cpg.TranslationManager.analyzeNonAsync(TranslationManager.kt:89)
	at de.fraunhofer.aisec.cpg.TranslationManager.analyze$lambda$0(TranslationManager.kt:71)
	at java.base/java.util.concurrent.CompletableFuture$AsyncSupply.run$$$capture(CompletableFuture.java:1768)
	at java.base/java.util.concurrent.CompletableFuture$AsyncSupply.run(CompletableFuture.java)
	at --- Async.Stack.Trace --- (captured by IntelliJ IDEA debugger)
	at java.base/java.util.concurrent.CompletableFuture$AsyncSupply.<init>(CompletableFuture.java:1754)
	at java.base/java.util.concurrent.CompletableFuture.asyncSupplyStage(CompletableFuture.java:1782)
	at java.base/java.util.concurrent.CompletableFuture.supplyAsync(CompletableFuture.java:1989)
	at de.fraunhofer.aisec.cpg.TranslationManager.analyze(TranslationManager.kt:71)
	at org.example.MainKt.main(Main.kt:35)
	... 1 more
Caused by: java.io.IOException: error=13, Permission denied
	at java.base/java.lang.ProcessImpl.forkAndExec(Native Method)
Caused by: java.io.IOException: error=13, Permission denied

	at java.base/java.lang.ProcessImpl.<init>(ProcessImpl.java:295)
	at java.base/java.lang.ProcessImpl.start(ProcessImpl.java:225)
	at java.base/java.lang.ProcessBuilder.start(ProcessBuilder.java:1126)
	... 18 more
```

# Unexpected behavior

*When the analysis terminates exception free but leads to unexpected graphs: Describe the difference between the generated graph and what you would have expected*

`/tmp/parser12814639995727740945` is an empty file and is not a correct TS parser. When I debug the parser, I find my Ubuntu returns `amd64` for `os.arch`. It turns out that

https://github.com/Fraunhofer-AISEC/cpg/blob/8834fb0e3a2556999ea46d0befb6873c0c9d7a1f/cpg-language-typescript/src/main/kotlin/de/fraunhofer/aisec/cpg/frontends/typescript/TypeScriptLanguageFrontend.kt#L91

cannot be a valid path since there is only `parser-linux-x86_64` in the TypeScript parser package.

# Code of interest

https://github.com/Fraunhofer-AISEC/cpg/blob/8834fb0e3a2556999ea46d0befb6873c0c9d7a1f/cpg-language-typescript/src/main/kotlin/de/fraunhofer/aisec/cpg/frontends/typescript/TypeScriptLanguageFrontend.kt#L77-L91

# Extensions to the library, e.g. added or changed LanguageFrontends or Passes

*changes to or added components*

added TypeScript frontend

# Changes to the configuration

*configuration print*

Nothing special

--- a/cpg-language-typescript/src/main/kotlin/de/fraunhofer/aisec/cpg/frontends/typescript/TypeScriptLanguageFrontend.kt
+++ b/cpg-language-typescript/src/main/kotlin/de/fraunhofer/aisec/cpg/frontends/typescript/TypeScriptLanguageFrontend.kt
@@ -74,7 +74,7 @@ class TypeScriptLanguageFrontend(
         private val parserFile: File = createTempFile("parser", "")
 
         init {
-            val arch = System.getProperty("os.arch")
+            val arch = System.getProperty("os.arch").replace("amd64", "x86_64")
             val os: String =
                 when {
                     System.getProperty("os.name").startsWith("Mac") -> {

